cheroot/server.py

```diff
@@ -1204,6 +1254,190 @@ def send_headers(self):
+        self.server = server
+        self.conn = conn
+        self.h_conn = h11.Connection(h11.SERVER)
+        self.ready = False
+        self.started_request = False
+        self.scheme = b'http'
+        if self.server.ssl_adapter is not None:
+            self.scheme = b'https'
+        # Use the lowest-common protocol in case read_request_line errors.
+        self.response_protocol = 'HTTP/1.0'
+        self.inheaders = {}
+
+        self.status = ''
+        self.outheaders = []
+        self.sent_headers = False
+        self.close_connection = self.__class__.close_connection
+        self.chunked_read = False
+        self.chunked_write = self.__class__.chunked_write
+        self.proxy_mode = proxy_mode
+        self.strict_mode = strict_mode
+
+    def _next_event(self):
+        while True:
+            event = self.h_conn.next_event()
+            if event is h11.NEED_DATA:
+                if self.h_conn.they_are_waiting_for_100_continue:
+                    go_ahead = h11.InformationalResponse(status_code=100)
+                    bytes_out = self.h_conn.send(go_ahead)
+                    self.conn.wfile.write(bytes_out)
+                self.h_conn.receive_data(self.conn.rfile.readline())
+                continue
+            return event
+
+    def parse_request(self):
+        """Parse the next HTTP request start-line and message-headers."""
+        req_line = self._next_event()
+        if isinstance(req_line, h11.Request):
+            self.started_request = True
+            self.uri = req_line.target
+            scheme, netloc, path, query, fragment = urllib.parse.urlsplit(self.uri)
+            self.path = path
+            self.qs = query
+            self.method = req_line.method
+            self.request_protocol = b"HTTP/%s" % req_line.http_version
+            self.response_protocol = b"HTTP/%s" % req_line.http_version
+            # TODO: oneliner-ify this
+            self.inheaders = {}
+            for header in req_line.headers:
+                self.inheaders[header[0]] = header[1]
+            self.ready = True
+        else:
+            # TODO
+            raise NotImplementedError("Only expecting Request object here")
+
+    def respond(self):
+        """Call the gateway and write its iterable output."""
+        mrbs = self.server.max_request_body_size
+        if not self.chunked_read:
+            cl = int(self.inheaders.get(b'Content-Length', 0))
+            if mrbs and mrbs < cl:
+                if not self.sent_headers:
+                    self.simple_response(
+                        '413 Request Entity Too Large',
+                        'The entity sent with the request exceeds the '
+                        'maximum allowed bytes.',
+                    )
+                return
+        self.rfile = HyperRFile(self.conn, self.h_conn)
+
+        self.server.gateway(self).respond()
+        self.ready and self.ensure_headers_sent()
+
+    def simple_response(self, status, msg=''):
```

Comment:
Usually, it's bad to use mutable defaults. Use `None` and assign the default in the beginning of the method
```suggestion
    def simple_response(self, status, msg=None):
```